workshop/5 - Security/README.md

```diff
@@ -0,0 +1,259 @@
+1. Enter `AppConfigSuperSecret` for the **Key** field.
+1. Under the **Key Vault** dropdown, select the Key Vault you created earlier. Make sure the correct resource group is selected.
+1. Once the Key Vault is selected, the **Secret** dropdown will be enabled. Select `SuperSecretSecret` from the dropdown and click **Apply**.
+1. The new configuration value will appear in the list. It will be indicated that it is a KeyVault reference by a lock icon.
+
+    ![screenshot of keyvault reference in app config](../images/5-Security/appconfig-keyvault.png)
+
+1. Now let's update the application to read the secret from Key Vault.
+1. Back to Visual Studio. Right-click on the solution in **Solution Explorer** and select **Manage NuGet Packages for Solution...**.
+1. Browse for **Azure.Security.KeyVault.Secrets** and install to the **AppConfigurationReader** project.
+1. Open the **Program.cs** file.
+1. Change the `builder.Configuration.AddAzureAppConfiguration` to look like this:
+
+    ```csharp
+    builder.Configuration.AddAzureAppConfiguration(options =>
+    {
+        options.Connect(new Uri(appConfigUrl), azureCredential)
+        .ConfigureKeyVault(kv =>
+        {
+            kv.SetCredential(azureCredential);
+        });
+    });
+    ```
+
+    This will configure our code to use Azure App Configuration to use the managed identity to authenticate to Key Vault.
+
+1. Open up the **Index.razor** file under the **Pages** directory on the **Relecloud Lite Web** project.
+1. Add in the following to the end of the file:
+
+    ```html
+    <h4>The KeyVault super duper secret is @Configuration["AppConfigSuperSecret"]</h4>
+    ```
+
+1. Run the application locally. You should see the message from Azure App Configuration and Azure KeyVault displayed.
+1. Using the PowerShell terminal, run the following command to deploy the application to Azure:
+
+    ```powershell
+    azd deploy
+    ```
+
+1. When the deployment is finished, browse to it and you should see the Azure App Service also has access to Azure Key Vault.
+
+## Review in the Reliable Web App's Code
+
+Let's review where how Managed Identity, Azure App Configuration, and Azure Key Vault are used in the Reliable Web App reference application.
+
+### User-assigned Managed Identity
+
+The Reliable Web App reference application uses User-Assigned Managed Identities rather than System-Assigned. The major difference is that User-Assigned Managed Identities are not tied to a specific Azure resource. This allows you to use the same identity across multiple resources. This is useful when you have multiple resources that need to access the same Azure resources.
+
+Of course, you also need to create the User-Assigned Managed Identity, then assign it to the various resources, then assign the appropriate permissions to the identity. This is all done through the Azure CLI, Azure portal, or even ARM or bicep templates when provisioning the Azure resources.
```

Comment:
Instead of "This is all done through..." - "You can assign permissions to a user-assigned managed identity through the Azure portal, CLI tools, ARM, Bicep, or other IaC tools such as Terraform when provisioning the Azure resources.  Use Azure Verified Modules to make this process easier."